js-core/homeworks/homework_6/src/main.js

```diff
@@ -0,0 +1,538 @@
+            return newItem;
+        });
+        const newStr = newArray.reverse().join(' ');
+        return newStr;
+    } else {
+        const myArray = str.split(' ');
+        const newArray = myArray.map( function invert(item) {
+            const newItem = item.split('').reverse().join('');
+            return newItem;
+        });
+        const newStr = newArray.join(' ');
+        return newStr;
+    }
+}
+
+console.log(reverseEachWord2('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM', true));
+// MOD eht fo snoitatnemelpmi tnereffid evah sresworb tnereffiD .MOD eht gnisu nigeb ot laiceps gnihtyna od ot evah t'nod uoY
+console.log(reverseEachWord('The Document Object Model (DOM) is a programming interface for HTML and XML documents', true));
+// stnemucod LMX dna LMTH rof ecafretni gnimmargorp a si )MOD( ledoM tcejbO tnemucoD ehT
+
+console.log(reverseEachWord2('Hi my Name is', false));
+// iH ym emaN si
+
+/* TASK 3
+ * Посчитайте сколько одинаковых слов в предложении.
+ * Функция должна возвращать объект у которого свойства
+ * это повторяющиеся слово,
+ * а значение это количество повторений
+ * */
+
+// Both - Java - and - Java - Script - is - programming - and - programming - OOPBased
+
+function wordCounter(sentence) {
+    let object = {};
+    let myArray = sentence.split(' ');
+    myArray.forEach(function(item) {
+        let elem = item;
+        let counter = 0;
+        myArray.forEach(function(item) {
+            let nextElem = item;
+            if (elem === nextElem) {
+                counter ++;
+            }
+            return counter;
+        });
+        object[elem] = counter;
+    });
+    return object;
+}
+
+console.log(
+    wordCounter('Both Java and Java Script is programming and programming OOPBased Language'),
+);
+// // /*
+//  {
+//  Both:1,
+//  Java:2,
+//  and:2,
+//  Script: 1,
+//  is: 1
+//  programming: 2
+//  OOPBased:1,
+//  Language:1
+//  }
+
+
+console.log(wordCounter('asd qwe asd'));
+/*
+ {
+ asd:2
+ qwe:1
+ }
+ * */
+
+console.log(wordCounter('url http url www url http'));
+
+/*
+ {
+ http:2,
+ url:3,
+ www:1
+ }
+ * */
+
+/*
+  TASK 4
+ // Функция принимает массив у которого есть свойста _id и Company.
+ // верните объект, у которого ключ это _id, а значение Company
+ */
+
+function createHashTags(arr) {
+    let object = {};
+    arr.forEach(function(item) {
+        object[item._id] = item.company;
+    });
+    return object;
+}
+
+console.log(createHashTags(listOfCompanys));
+
+//{"584babb6eeb4137cf14c37a3":"ASIMILINE", '584babb6eeb4137cf14c37a3':'Company2', }
+
+// @ SUPER
+/*
+ *
+ * TASK 1
+ * Выведите уникальные значения
+ *
+ * */
+
+function uniqueElements(arr) {
+    let newArr = [];
+    newArr.push(arr[0]);
+    arr.forEach(function (item) {
+        let elem = item;
+        function compare(item) {
+            return elem !== item;
+        }
+        if (newArr.every(compare)) {
```

Comment:
interesting solution :) a like it.

But I think we can improve 1 part of it.
I guess we can omit using ".every" and find something a bit better.

Also it worth to think about **.filter**, because you have to make given array smaller